services/management/ia/templates_service.go

```diff
@@ -151,6 +162,126 @@ func (svc *TemplatesService) loadFile(ctx context.Context, file string) ([]saas.
 	return rules, nil
 }
 
+type ruleFile struct {
+	Group []ruleGroup `yaml:"groups"`
+}
+
+type ruleGroup struct {
+	Name  string `yaml:"name"`
+	Rules []rule `yaml:"rules"`
+}
+
+type rule struct {
+	Alert       string              `yaml:"alert"` // same as alert name in template file
+	Expr        string              `yaml:"expr"`
+	Duration    promconfig.Duration `yaml:"for"`
+	Labels      map[string]string   `yaml:"labels,omitempty"`
+	Annotations map[string]string   `yaml:"annotations,omitempty"`
+}
+
+// converts an alert template rule to a rule file. generates one file per rule.
+func (svc *TemplatesService) convertTemplates(ctx context.Context) error {
+	templates := svc.getCollected(ctx)
+	for _, template := range templates {
+		r := rule{
+			Alert:       template.Name,
+			Duration:    template.For,
+			Labels:      make(map[string]string, len(template.Labels)),
+			Annotations: make(map[string]string, len(template.Annotations)),
+		}
+
+		data := make(map[string]string, len(template.Params))
+		for _, param := range template.Params {
+			data[param.Name] = fmt.Sprint(param.Value)
+		}
+
+		var buf bytes.Buffer
+		t, err := newParamTemplate().Parse(template.Expr)
+		if err != nil {
+			return errors.Wrap(err, "failed to convert rule template")
+		}
+		if err = t.Execute(&buf, data); err != nil {
+			return errors.Wrap(err, "failed to convert rule template")
+		}
+		r.Expr = buf.String()
+
+		err = transformMaps(template.Labels, r.Labels, data)
+		if err != nil {
+			return errors.Wrap(err, "failed to convert rule template")
+		}
+
+		// add parameters to labels
+		for _, p := range template.Params {
+			r.Labels[p.Name] = fmt.Sprint(p.Value)
+		}
+
+		// add special labels
+		r.Labels["ia"] = "1"
+		r.Labels["severity"] = template.Severity.String()
+
+		err = transformMaps(template.Annotations, r.Annotations, data)
+		if err != nil {
+			return errors.Wrap(err, "failed to convert rule template")
+		}
+
+		rf := ruleFile{
+			Group: []ruleGroup{{
+				Name:  "PMM Server Integrated Alerting",
+				Rules: []rule{r},
+			}},
+		}
+
+		err = dumpRule(rf)
+		if err != nil {
+			return errors.Wrap(err, "failed to dump alert rules")
+		}
+	}
+	return nil
+}
+
+// fills templates found in label and annotaions with values.
+func transformMaps(src map[string]string, dest map[string]string, data map[string]string) error {
+	var buf bytes.Buffer
+
+	for k, v := range src {
+		buf.Reset()
+		t, err := newParamTemplate().Parse(v)
+		if err != nil {
+			return err
+		}
+		if err = t.Execute(&buf, data); err != nil {
+			return err
+		}
+		dest[k] = buf.String()
+	}
+	return nil
+}
+
+// dump the transformed IA rules to a file.
+func dumpRule(rule ruleFile) error {
+	b, err := yaml.Marshal(rule)
+	if err != nil {
+		return errors.Errorf("failed to marshal rule %s", err)
+	}
+	b = append([]byte("---\n"), b...)
+
+	filepath := ruleFileDir + rule.Group[0].Rules[0].Alert + ".yml"
```

Comment:
`filepath` variable collides with the package name, better to name it just `path`